src/pages/CustomerInvoicePage.js

```diff
@@ -35,56 +34,32 @@ const MODAL_KEYS = {
 export class CustomerInvoicePage extends GenericPage {
   constructor(props) {
     super(props);
-    this.state.sortBy = 'itemName';
-    this.state.columns = [
-      {
-        key: 'itemCode',
-        width: 2,
-        title: tableStrings.item_code,
-        sortable: true,
-      },
-      {
-        key: 'itemName',
-        width: 4,
-        title: tableStrings.item_name,
-        sortable: true,
-      },
-      {
-        key: 'availableQuantity',
-        width: 2,
-        title: tableStrings.available_stock,
-        sortable: true,
-        alignText: 'right',
-      },
-      {
-        key: 'totalQuantity',
-        width: 2,
-        title: tableStrings.quantity,
-        sortable: true,
-        alignText: 'right',
-      },
-      {
-        key: 'remove',
-        width: 1,
-        title: tableStrings.remove,
-        alignText: 'center',
-      },
-    ];
-    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
-    this.finalisableDataType = 'Transaction';
-    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
-    this.onAddMasterItems = this.onAddMasterItems.bind(this);
-    this.openItemSelector = this.openItemSelector.bind(this);
-    this.openCommentEditor = this.openCommentEditor.bind(this);
-    this.openTheirRefEditor = this.openTheirRefEditor.bind(this);
-    this.getModalTitle = this.getModalTitle.bind(this);
-    this.renderPageInfo = this.renderPageInfo.bind(this);
+    this.state = {
+      selection: [],
+      modalKey: null,
+      modalIsOpen: false,
+    };
+    this.dataFilters = {
+      searchTerm: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    autobind(this);
+  }
+
+  updateDataFilters(newSearchTerm, newSortBy, newIsAscending) {
+    // We use != null, which checks for both null or undefined (undefined coerces to null)
+    if (newSearchTerm != null) this.dataFilters.searchTerm = newSearchTerm;
```

Comment:
One things that is bothers me with refactor to composition.. duplicate updateDataFilters and very similar this.dataFilters declaration.